AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -40,39 +41,110 @@ def __init__(self, sc_editor):
         """
-        function for checking the provided variable type for validity
+        function for checking the provided variable type validity
 
         returns: None
         """
         assert type(new_variable_type) is str, "Wrong parameter type provided. new_variable_type was not str"
         assert new_variable_type in VARIABLE_TYPES_DICT, \
             "Wrong type provided. new_variable_type is not a valid type"
 
-    def create_new_variable(self, new_variable_type: str) -> None:
+    def __click_on_add_variable_button(self) -> None:
         """
-        Function for adding a new variable to the variable manager's list
+        helper function for adding a new variable. Clicks on the 'Create Variable' button and waits for the UI to render
 
-        param new_variable_type: the type of variable you want to create provided as a string. Is case-sensitive!
-
-        returns: None
+        returns None
         """
-        self.__validate_new_variable(new_variable_type)
-
         add_new_variable_button = self.variable_manager.findChild(QtWidgets.QPushButton, ADD_BUTTON_QT)
         add_new_variable_button.click()  # Click on Create Variable button
 
         helper.wait_for_condition((
             lambda: self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT) is not None),
             WAIT_TIME_SEC_3)
 
+    def __select_variable_type_from_create_menu(self, new_variable_type: str) -> None:
+        """
+        helper function for adding a new variable. selects the desired variable type from the menu that renders after
+        the create variable button is clicked. waits for the UI to respond after making a selection
+
+        params: The type of variable to create as a string. ie "Boolean"
+
+        returns none
+        """
+
         # Select variable type
         table_view = self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT)
         variable_entry = pyside_utils.find_child_by_pattern(table_view, new_variable_type)
 
-        # Click on it to create variable
+        # Click on it to create variable and wait 1 second for the UI to respond
         pyside_utils.item_view_index_mouse_click(table_view, variable_entry)
+        helper.wait_for_condition(lambda: True is False, WAIT_TIME_SEC_1)
+
+    def __set_new_variable_name(self, new_variable_name: str):
+
+        """
+        helper function for adding a new variable. sets the new variable's name. waits for the UI to respond afterwards
+
+        returns Qwidget with the graph variables that have been created
+        """
+
+        # Set the new variable's name. Wait 1 second for the UI to respond
+        graph_vars_page = self.variable_manager.findChild(QtWidgets.QWidget, GRAPH_VARIABLES_PAGE_QT)
+        line_edit = graph_vars_page.findChild(QtWidgets.QLineEdit, "")
+        line_edit.insert(new_variable_name)
+        helper.wait_for_condition(lambda: True is False, WAIT_TIME_SEC_1)
```

Comment:
Another alarming wait.

Reply:
replaced this with a check against the text that's currently in that line edit field. helper wait for condition waits 3 seconds